x-pack/plugins/security_solution/public/management/pages/policy/view/trusted_apps/flyout/policy_trusted_apps_flyout.test.tsx

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { PolicyTrustedAppsFlyout } from './policy_trusted_apps_flyout';
+import * as reactTestingLibrary from '@testing-library/react';
+import { fireEvent } from '@testing-library/dom';
+import {
+  AppContextTestRender,
+  createAppRootMockRenderer,
+} from '../../../../../../common/mock/endpoint';
+import { MiddlewareActionSpyHelper } from '../../../../../../common/store/test_utils';
+
+import { TrustedAppsHttpService } from '../../../../trusted_apps/service';
+import { PolicyDetailsState } from '../../../types';
+import { getMockCreateResponse, getMockListResponse } from '../../../test_utils';
+import { createLoadedResourceState } from '../../../../../state';
+
+jest.mock('../../../../trusted_apps/service');
+
+let mockedContext: AppContextTestRender;
+let waitForAction: MiddlewareActionSpyHelper['waitForAction'];
+let render: () => ReturnType<AppContextTestRender['render']>;
+const act = reactTestingLibrary.act;
+const TrustedAppsHttpServiceMock = TrustedAppsHttpService as jest.Mock;
+let getState: () => PolicyDetailsState;
+
+describe('Policy trusted apps flyout', () => {
+  beforeEach(() => {
+    TrustedAppsHttpServiceMock.mockImplementation(() => {
+      return {
+        getTrustedAppsList: () => getMockListResponse(),
+        updateTrustedApp: () => ({
+          data: getMockCreateResponse(),
+        }),
+      };
+    });
+  });
+  beforeEach(() => {
+    mockedContext = createAppRootMockRenderer();
+    waitForAction = mockedContext.middlewareSpy.waitForAction;
+    getState = () => mockedContext.store.getState().management.policyDetails;
+    render = () => mockedContext.render(<PolicyTrustedAppsFlyout />);
+  });
+
+  afterEach(() => reactTestingLibrary.cleanup());
+
+  it('should renders flyout open correctly without available data', async () => {
+    const waitAvailableListExist = waitForAction('policyArtifactsAvailableListExistDataChanged');
+
+    TrustedAppsHttpServiceMock.mockImplementation(() => {
+      return {
+        getTrustedAppsList: () => ({ data: [] }),
+      };
+    });
+    const component = render();
+    mockedContext.store.dispatch({
+      type: 'userChangedUrl',
+      payload: {
+        pathname: '/administration/policy/1234/trustedApps',
```

Comment:
I was going to suggest (for here and other cases below) to use the `get**Path` methods from the `routing` module under our common folder so that they would not be hard coded, but I see you are doing a direct dispatch to the store. an alternative would be to use the `mockedContext.history.push` method and there you can just use the path getters.